src/transformState.js

```diff
@@ -1,11 +1,34 @@
-'use strict';
+"use strict";
 
 /**
  * @param {Object} state
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const obj of actions) {
+    switch (obj.type) {
+      case "addProperties":
+        Object.assign(state, obj.extraData);
+
+        break;
+
+      case "removeProperties":
+        for (const key of obj.keysToRemove) {
+          delete state[key];
+        }
+
+        break;
+
+      case "clear":
+        for (const del in state) {
+          delete state[del];
+        }
+
+        break;
```

Comment:
But `default` case still need. Always use `default` case for each switch construction